### PR TITLE
Fix Instance Pool Instance List

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Fix Instance Pool Instance list

### CODE REVIEW

**Comment by falzm** [comment]
Acceptance tests fail:

```
=== RUN   TestAccResourceInstancePool
--- FAIL: TestAccResourceInstancePool (176.91s)
    testing.go:569: Step 0 error: Check failed: Check 3/3 error: expected attribute "virtual_machines" not found in map
```

File: exoscale/resource_exoscale_instance_pool_test.go [hunk 1]
```diff
@@ -236,7 +238,7 @@ func testAccCheckResourceInstancePoolDestroy(s *terraform.State) error {
 		}
 
 		// this time.Sleep() is here to prevent race condition when
-		// an instance pool is destroyed, to wait till instance pool state chage
+		// an instance pool is destroyed, to wait till instance pool state change
```
Comment (falzm) [inline exoscale/resource_exoscale_instance_pool_test.go]:
```suggestion
		// an instance pool is destroyed, to wait till instance pool state changes
```

File: exoscale/validation.go [hunk 2]
```diff
@@ -48,6 +48,24 @@ func ValidateRegexp(pattern string) schema.SchemaValidateFunc {
 	}
 }
 
+// ValidateSliceNotEmpty validates that the given field is a slice and not empty.
```
Comment (falzm) [inline exoscale/validation.go]:
```suggestion
// ValidateSliceNotEmpty validates that the given field is a non-empty slice.
```

Comment (pierre-emmanuelJ) [reply exoscale/validation.go]: I did a change before your suggestion:
https://github.com/terraform-providers/terraform-provider-exoscale/pull/53/commits/5e837c98cd4035d0a1cd29d42092702cf2a37de8

**Comment by pierre-emmanuelJ** [comment]
```
=== RUN   TestAccResourceInstancePool
--- PASS: TestAccResourceInstancePool (162.51s)
PASS
ok  	github.com/terraform-providers/terraform-provider-exoscale/exoscale	162.543s
```

**Comment by pierre-emmanuelJ** [comment]
Tests on latest commit:
```
=== RUN   TestAccResourceInstancePool


--- PASS: TestAccResourceInstancePool (172.04s)
PASS
ok  	github.com/terraform-providers/terraform-provider-exoscale/exoscale	172.061s
```

File: exoscale/validation.go [hunk 3]
```diff
@@ -29,6 +29,24 @@ func ValidateString(str string) schema.SchemaValidateFunc {
 	}
 }
 
+// ValidateStringNot validates that the given field is a string and not matches the expected value
```
Comment (falzm) [inline exoscale/validation.go]:
```suggestion
// ValidateStringNot validates that the given field is a string that doesn't match the specified value.
```

**Review by falzm — APPROVED** [review]
LGTM 👍

**Pull request merged**